src/main/java/aztech/modern_industrialization/machines/MachineBlockEntity.java

```diff
@@ -176,10 +139,10 @@ public final void writeScreenOpeningData(RegistryFriendlyByteBuf buf) {
         inv.fluidPositions.write(buf);
         buf.writeInt(guiComponents.size());
         // Write components
-        for (GuiComponent.Server component : guiComponents) {
+        guiComponents.forEach(component -> {
```

Comment:
Please undo all the `forEach` conversions, they shouldn't be necessary when the component storage implements `Iterable<C>`.